internal/controller/apiextensions/conditions/conditions.go

```diff
@@ -0,0 +1,224 @@
+package conditions
+
+import (
+	"errors"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composite"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	// conditions that should always be present and are controlled by the system
+	systemConditionTypes = map[xpv1.ConditionType]struct{}{
+		xpv1.TypeReady:  {},
+		xpv1.TypeSynced: {},
+	}
+)
```

Comment:
Nit: If we don't think this will grow much above two types I'd prefer to just hard code them rather than use global state. e.g. Inside a `Reserved(t xpv1.ConditionType) bool` function that uses a switch to return true for system conditions.